### PR TITLE
Use dynamic redefs throughout tests

### PR DESCRIPTION
**Pull request by crisptrutski**
It's very convenient to stub out functions in tests, and we do this a lot. Unfortunately the way we were doing it was no thread-safe, which is a bother for tests that could otherwise be run in parallel.

This PR replaces all our test usage with our custom thread-safe version from https://github.com/metabase/metabase/pull/38620

Hopefully it is discoverable enough now and people internalize reaching for it. I'm going to be optimistic / lazy and skip creating a linter for this.

### CODE REVIEW

**Comment by crisptrutski** [comment]
Whitespace was a pain. I've probably messed up a few spots, but I got most of them. I think it's OK for boy scouts to fix up the occasional alignment as they come across it.

File: test/metabase/driver/sql_jdbc/connection_test.clj [hunk 1]
```diff
@@ -48,7 +48,7 @@
             original-destroy   @#'sql-jdbc.conn/destroy-pool!
             connection-details {:db "mem:connection_test"}
             spec               (mdb.spec/spec :h2 connection-details)]
-        (with-redefs [sql-jdbc.conn/destroy-pool! (fn [id destroyed-spec]
+        (mt/with-dynamic-redefs [sql-jdbc.conn/destroy-pool! (fn [id destroyed-spec]
```
Comment (calherries) [inline test/metabase/driver/sql_jdbc/connection_test.clj:51]: You broke formatting here

File: test/metabase/driver/sql/query_processor_test.clj [hunk 2]
```diff
@@ -243,7 +243,7 @@
 
 (deftest adjust-start-of-week-test
   (driver/with-driver :h2
-    (with-redefs [driver/db-start-of-week   (constantly :monday)
+    (mt/with-dynamic-redefs [driver/db-start-of-week   (constantly :monday)
```
Comment (calherries) [inline test/metabase/driver/sql/query_processor_test.clj:246]: formatting

**Review by calherries — APPROVED** [review]
Nice, this will help with discoverability.

There is a lot of broken formatting and I gave up commenting on them so I don't blame you for missing them :D But you should clean these up before merging

**Comment by calherries** [comment]
> I think it's OK for boy scouts to fix up the occasional alignment as they come across it.

I don't think that makes sense from an efficiency pov. It's much less effort to have everything updated in one go than a thousand times separately.

**Comment by calherries** [comment]
It's a shame we don't have an auto-formatter for these things though

**Comment by calherries** [comment]
@crisptrutski did you want to close this?

**Comment by crisptrutski** [comment]
Yeah, given that the issues went deeper than just formatting, I've lost the energy for a big-bang refactor.

**Pull request closed without merging**